### ISSUE TITLE
Request for help: Error reading parquet files

### ISSUE THREAD

**Issue by assayire** [jerolba/parquet-for-java-posts#1]
Hello @jerolba,

I am reading/following your posts ([➤](https://www.jeronimo.dev/working-with-parquet-files-in-java-using-avro/)) on working with parquet files. I have to say yours is the only source of consumable/useful information I have found on the topic. So, a big thanks.

I am able to successfully write a parquet file. However, when reading the same file, I am getting the following exception. I have tried a few different things but no luck. For instance, I have tried to write just one record (instead of a list) with just a couple of numeric fields, still the same issue.

```
Exception in thread "main" java.lang.RuntimeException: shaded.parquet.com.fasterxml.jackson.databind.exc.InvalidDefinitionException: No serializer found for class org.apache.parquet.schema.LogicalTypeAnnotation$StringLogicalTypeAnnotation and no properties discovered to create BeanSerializer (to avoid exception, disable SerializationFeature.FAIL_ON_EMPTY_BEANS) (through reference chain: org.apache.parquet.hadoop.metadata.ParquetMetadata["fileMetaData"]->org.apache.parquet.hadoop.metadata.FileMetaData["schema"]->org.apache.parquet.schema.MessageType["fields"]->java.util.ArrayList[0]->org.apache.parquet.schema.PrimitiveType["logicalTypeAnnotation"])
	at org.apache.parquet.hadoop.metadata.ParquetMetadata.toJSON(ParquetMetadata.java:68)
	at org.apache.parquet.hadoop.metadata.ParquetMetadata.toPrettyJSON(ParquetMetadata.java:48)
	at org.apache.parquet.format.converter.ParquetMetadataConverter.readParquetMetadata(ParquetMetadataConverter.java:1592)
	at org.apache.parquet.hadoop.ParquetFileReader.readFooter(ParquetFileReader.java:629)
	at org.apache.parquet.hadoop.ParquetFileReader.<init>(ParquetFileReader.java:902)
	at org.apache.parquet.hadoop.ParquetFileReader.open(ParquetFileReader.java:710)
	at org.apache.parquet.hadoop.ParquetReader.initReader(ParquetReader.java:166)
	at org.apache.parquet.hadoop.ParquetReader.read(ParquetReader.java:139)
	at park.Runner.parseOrgs(Runner.java:185)
```

I was wondering if you have come across this issue. Or can spot anything from the error. _Any help in this regard is highly appreciated_.

P.S: If you like to take a look at my code, [**this**](https://github.com/assayire/park) is the repo.

**Comment by jerolba** [comment]
Hi assayire.

Thanks for your feedback, I decided to write those articles because of that :)

You are using "latest" Parquet version that has a reported issue:

https://issues.apache.org/jira/browse/PARQUET-2468

Yesterday a new version was released fixing the issue:

https://parquet.apache.org/blog/2024/06/16/1.14.1/

FYI. I've created a library that wraps Parquet library that doesn't depend of Avro or Protobuf, simplifying all the configuration process. It's open source and available in maven central:

https://github.com/jerolba/parquet-carpet

I invite you to test it :)
